Assets/Scripts/Structres/ProjectileTower.cs

```diff
@@ -1,26 +1,62 @@
 using UnityEngine;
 using FishNet;
 using FishNet.Object;
+using Examen.Proximity;
+using System.Collections.Generic;
 
 namespace Examen.Structures
 {
+    [RequireComponent(typeof(ProximityAgent))]
     public class ProjectileTower : NetworkBehaviour
     {
         [SerializeField] private Transform _barrel;
         [SerializeField] private Transform _firePoint;
         [SerializeField] private NetworkObject projectilePrefab;
         [SerializeField] private float _shootInterval;
 
+        private ProximityAgent _proximityAgent;
         private Transform _boss;
         private float _timer = 0f;
 
-        private void Awake() => _boss = GameObject.Find("Boss").transform; // TODO: Replace with a more reliable way to find the boss
+        private void Awake() => _proximityAgent = GetComponent<ProximityAgent>();
+
+        private void OnEnable() =>  _proximityAgent.OnProximityDataReceived += OnProximityDataReceived;
+
+        private void OnDisable() => _proximityAgent.OnProximityDataReceived -= OnProximityDataReceived;
+
+        private void OnProximityDataReceived(HashSet<ProximityAgent> nearbyAgents)
+        {
+            Debug.LogError("Proximity data received");
+            Debug.LogError(nearbyAgents.Count);
+
+            if (nearbyAgents.Count == 0)
+                return;
+
+            Transform closestAgent = null;
+            float closestDistance = float.MaxValue;
+
+            foreach (ProximityAgent agent in nearbyAgents)
+            {
+                float distance = (agent.transform.position - transform.position).sqrMagnitude;
+
+                if (distance < closestDistance)
+                {
+                    closestDistance = distance;
+                    closestAgent = agent.transform;
+                }
+            }
+
+            _boss = closestAgent;
+        }
 
         private void FixedUpdate()
         {
             if (!IsServer)
                 return;
 
+            if (_boss == null)
+                return;
```

Comment:
Can this if and the if above made into one if statement?

Reply:
No because than each client will return because of the is server check so only the server needs to do the null check. If its in one the client needs to do both